TypeScript (#1819)
enkidevs/curriculum: typescript/introduction/intro-typescript/hello-typescript.md

---
author: kapnobatai136

type: normal

category: must-know

aspects:
  - introduction

links:
  - '[ts-node](https://www.npmjs.com/package/ts-node){website}'

---

# Hello TypeScript

---
## Content

We have promised you some TypeScript code, and here it is:

```ts
// hello.ts
const hello = (name: string) => {
  console.log(`Hello ${name}!`)
}

let myName = 'Andrei';
hello(myName);
```

> When running `tsc hello.ts` in the shell, a new file `hello.js` is created, and these are its contents:

Reviewer: ```suggestion
When running `tsc hello.ts` in the shell, TypeScript code is converted into JavaScript and a new file `hello.js` is created:
```

```js
// hello.js
var hello = function (name) {
    console.log("Hello " + name + "!");
};
var myName = 'Andrei';
hello(myName);
```

> Depending on your application, this might be enough or you might want to run the `.js` file using node. In the latter case, instead of running two separate commands, you can install the [ts-node package](https://www.npmjs.com/package/ts-node) and let it do the work for you.

Reviewer: Add this as a footnote?

```suggestion
Depending on your application, this might be enough or you might want to run the `.js` file using node[1]
```

Reviewer: Hmmm actually nevermind, I guess it's explained below as well.

> Depending on your application, this might be enough or you might want to run the `.js` file using node. In the latter case, instead of running two separate commands, you can install the [ts-node package](https://www.npmjs.com/package/ts-node) and let it do the work for you.

Reviewer: ```suggestion
Once you have the JavaScript code, you can run it wherever JavaScript can run (any Node.js or browser environment).

To convert TypeScript to JavaScript and run the generated JavaScript as a single step, we can use the [ts-node package](https://www.npmjs.com/package/ts-node).

```sh
# convert hello.ts to from
# Typescript to JavaScript
# and run the created JavaScript
ts-node hello.ts
```
```

> With this package installed, you can run `ts-node hello.ts` and it will compile the TypeScript code, and run the resulting JavaScript file.

Reviewer: If we apply something like I suggested above, we can remove this sentence

---
## Practice

> To **compile** TypeScript code, you have to run the following command:

Reviewer: ```suggestion
To convert TypeScript code into JavaScript, we can use the following command:
```

```bash
??? file.ts
```

To **compile and run** a JavaScript file, you have to run the following commands:

```bash

> tsc file.ts

Reviewer: Doesn't this give out the answer for the above? Shouldn't this question ask about `ts-node`?

---
author: kapnobatai136

type: normal

category: must-know

aspects:
  - introduction

links:
  - '[ts-node](https://www.npmjs.com/package/ts-node){website}'

---

# Hello TypeScript

---
## Content

We promised you some TypeScript code, and here it is:

```ts
// hello.ts
const hello = (name: string) => {
  console.log(`Hello ${name}!`)
}

hello('Andrei');
```

When running `tsc hello.ts` in the shell, TypeScript code is converted (compiled) into JavaScript and a new file `hello.js` is created:

```js
// hello.js
var hello = function (name) {
    console.log("Hello " + name + "!");
};
hello('Andrei');
```

Once you have the JavaScript code, you can run it wherever JavaScript can run (any Node.js or browser environment).

> To compile TypeScript to JavaScript and run the generated JavaScript as a single step, we can use the [ts-node package](https://www.npmjs.com/package/ts-node).

Reviewer 2: not sure if it's worth linkinng out of the app for an `npm` package

Author: > not sure if it's worth linkinng out of the app for an npm package
why not? you can still check out what the package does and some examples on mobile + you can bookmark it in your browser (or find it in history)

Author: > I think this makes more sense in the previous insight

previous insight is already quite big (sitting at 160 words). moving everything related to `ts-node` will push it to the limit, and we're trying to stay await from big insights.

also, you don't need `ts-node` to compile/use typescript. it is more of a nice-to-have

Reviewer: I vote that we keep this, I enjoyed the fact that I can link to the package and read about it.

Reviewer 3: imo ts-node is not relevant for the "hello world" example

Reviewer 4: It is, its an alternative to `tsc` for running TS

```sh
# convert hello.ts from
# Typescript to JavaScript
# and run the created JavaScript
ts-node hello.ts
```

---
## Practice

> To **compile** a TypeScript file called `enki.ts` into a JavaScript `enki.js`, you have to run the following command:

Reviewer: Yes it is, but it doesn't produce an `enki.js` file

> To **compile** a TypeScript file called `enki.ts` into a JavaScript `enki.js`, you have to run the following command:

Reviewer 3: but it's still compilation - I mean that the question can be confusing

```bash
??? file.ts
```

* tsc
* ts-node
* ts
* typescript

---
## Revision

> To **compile and run** a TypeScript file called `enki.ts`, you have to run the following commands:

Reviewer: ```suggestion
To **compile and run** a TypeScript file called `enki.ts`, you can run the following command:
```
